README.md

```diff
@@ -717,6 +773,105 @@ codePush.checkForUpdate()
+
+2. **Silent sync everytime the app resumes**. Same as 1, except we check for updates, or apply an update if one exists every time the app returns to the foreground after being "backgrounded".
+
+    ```javascript
+    // Sync for updates everytime the app resumes.
+    @codePushify({ syncMode: codePush.SyncMode.ON_APP_RESUME, installMode: codePush.InstallMode.ON_NEXT_RESUME })
+    class MyApp extends Component {}
+    ```
+
+3. **Interactive**. When an update is available, prompt the end user for permission before downloading it, and then immediately apply the update. If an update was released using the `mandatory` flag, the end user would still be notified about the update, but they wouldn't have the choice to ignore it.
+
+    ```javascript
+    // Active update, which lets the end user know
+    // about each update, and displays it to them
+    // immediately after downloading it
+    @codePushify({ updateDialog: true, installMode: codePush.InstallMode.IMMEDIATE })
+    class MyApp extends Component {}
+    ```
+
+4. **Log/display progress**. While the app is syncing with the server for updates, make use of the `codePushStatusDidChange` and/or `codePushDownloadDidProgress` event hooks to log down the different stages of this process, or even display a progress bar to the user.
+
+    ```javascript
+    // Make use of the event hooks to keep track of 
+    // the different stages of the sync process.
+    @codePushify()
+    class MyApp extends Component {
+        codePushStatusDidChange(status) {
+            switch(syncStatus) {
+                case codePush.SyncStatus.CHECKING_FOR_UPDATE:
+                    console.log("Checking for updates.");
+                    break;
+                case CodePush.SyncStatus.DOWNLOADING_PACKAGE:
+                    console.log("Downloading package.");
+                    break;
+                case CodePush.SyncStatus.INSTALLING_UPDATE:
+                    console.log("Installing update.");
+                    break;
+                case CodePush.SyncStatus.UP_TO_DATE:
+                    console.log("Installing update.");
+                    break;
+                case CodePush.SyncStatus.UPDATE_INSTALLED:
+                    console.log("Update installed.");
+                    break;
+            }
+        }
+        
+        codePushDownloadDidProgress(progress) {
+            console.log(progess.receivedBytes + " of " + progress.totalBytes + " received.");
+        }
+    }
+    ```
+
+##### CodePushOptions
+
+The `codePushify` decorator accepts an "options" object that allows you to customize numerous aspects of the default behavior mentioned above:
+
+* __syncMode__ *(codePush.SyncMode)* - Specifies when you would like to check for updates. Defaults to `codePush.SyncMode.ON_APP_START`. Refer to the [`SyncMode`](#syncmode) enum reference for a description of the available options and what they do.
+
+* __deploymentKey__ *(String)* - Refer to [`SyncOptions`](#syncoptions).
```

Comment:
If we're intending for the decorator to be the standard prescription, maybe it would be better to describe the options here, and link to this from syncOptions rather than the other way around